packages/client/auth-core/src/contexts/authlogin.tsx

```diff
@@ -0,0 +1,109 @@
+'use client'
+
+import AuthModal from "@/components/AuthModal";
+import { CrossmintService, FetchCrossmintParams } from "@/services/CrossmintService";
+import { CrossmintEnvironment } from "@/utils";
+import { ReactNode, createContext, useCallback, useContext, useEffect, useMemo, useRef, useState } from "react";
+
+type AuthContextType = {
+    user: User | null;
+    login: () => void;
+    logout: () => void;
+    fetchCrossmintAPI: (fetchCrossmintParams: FetchCrossmintParams) => Promise<any> | void;
```

Comment:
As the response depends on the endpoint, we cannot know beforehand what we expect to receive. We could use a generic type, where the client needs to specify it, but I would let that as an improvement as it would mean using a seemed strategy to the one used in the window package.

BTW, I assume this because on the CrossmintService used on AA we return a promise<any> too, so I think we can live with this just for the moment.

Reply:
The `fetchCrossmintAPI` was thought for internal purposes. As of now, this will be used by external devs, this function doesn't make sense anymore